core/amber/src/main/python/core/storage/runnables/port_result_writer.py

```diff
@@ -0,0 +1,35 @@
+import queue
+import threading
+
+from core.models import Tuple
+from core.storage.model.buffered_item_writer import BufferedItemWriter
+from core.util import Stoppable
+from core.util.runnable.runnable import Runnable
+
+
+class PortResultWriter(Runnable, Stoppable):
+    def __init__(self, writer: BufferedItemWriter):
+        self.writer: BufferedItemWriter = writer
+        self.queue = queue.Queue()
+        self.stopped = False
+        self.stop_event = threading.Event()
+
+    def put_tuple(self, amber_tuple: Tuple):
+        assert not self.stopped, "Cannot put tuple after termination."
+        self.queue.put(amber_tuple)
+
+    def run(self) -> None:
+        internal_stop = False
+        while not internal_stop:
```

Comment:
try to extend `StoppableQueueBlockingRunnable` so that you don't have to re-invent the wheels to stop it.

Reply:
Done